webserver/Controllers/ClientController.cs

```diff
@@ -0,0 +1,149 @@
+        _context = context;
+        _userManager = userManager;
+        _roleManager = roleManager;
+    }
+
+    [HttpGet("unique/{id}")]
+    public IActionResult ReadClient(string id) {
+
+        var client = _context.Clients.Find(id);
+        if(client==null){
+            return NotFound();
+        }
+
+        var response = JsonConvert.SerializeObject((ClientDTO)client);
+
+        return Ok(response);
+    }
+
+    [HttpGet]
+    public async Task<IActionResult> ReadClients(string? username, int? offset, int? limit, string? sort) {
+
+        if(limit<1){
+            return BadRequest("Results amount are limited to less than 1");
+        }
+
+        var clients = _context.Clients.AsQueryable();
+
+        if(!string.IsNullOrEmpty(username)){
+            clients = clients.Where(client => client.UserName!.Contains(username));
+        }
+
+
+        if(!string.IsNullOrEmpty(sort)){
+            sort = sort.ToLower();
+            switch (sort) {
+                case "name":
+                    clients = clients.OrderBy(re => re.UserName);
+                    break;
+            }
+        }
+
+        if(offset.HasValue){
+            clients = clients.Skip(offset.Value);
+        }
+        if(limit.HasValue){
+            clients = clients.Take(limit.Value);
+        }
+
+        var resultArray = await clients.ToArrayAsync();
+        var resultDtoArray = resultArray.Select(c=>(ClientDTO)c).ToArray();
+        
+        if(resultDtoArray==null || resultDtoArray.Length==0){
+            return NotFound();
+        }
+        
+        var response = JsonConvert.SerializeObject(resultDtoArray);
+
+        return Ok(response);
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> CreateClient([FromBody] ClientDTO clientDto, string password) {
+
+        // Check if the email is already registered
+        var existingEmail = await _userManager.FindByEmailAsync(clientDto.Email);
+        if (existingEmail != null) {
+            return BadRequest("Email already registered!");
+        }
+
+        var existingUsername = await _userManager.FindByNameAsync(clientDto.UserName);
+        if (existingUsername != null) {
+            return BadRequest("UserName already registered!");
+        }
+
+        Client client = (Client)clientDto;
+
+        var result = await _userManager.CreateAsync(client, password);
+
+        if (result.Succeeded) {
+
+            var roleExists = await _roleManager.RoleExistsAsync(Common.Client_Role);
+            if (!roleExists) {
+                await _roleManager.CreateAsync(new IdentityRole(Common.Client_Role));
+            }
+
+            await _userManager.AddToRoleAsync(client, Common.Client_Role);
+
+            return CreatedAtAction(nameof(CreateClient), (ClientDTO)client);
+        }else{
+            return StatusCode(500, "Internal Server Error: Register Client Unsuccessful");
+        }
```

Comment:
You can write a cleaner error validation.
Read more: https://www.geeksforgeeks.org/writing-clean-else-statements/

```suggestion
        if (!result.Succeeded) {
            return StatusCode(500, "Internal Server Error: Register Client Unsuccessful");
        }
        
        var roleExists = await _roleManager.RoleExistsAsync(Common.Client_Role);
        if (!roleExists) {
            await _roleManager.CreateAsync(new IdentityRole(Common.Client_Role));
        }

        await _userManager.AddToRoleAsync(client, Common.Client_Role);

        return CreatedAtAction(nameof(CreateClient), (ClientDTO)client);
```